pkg/resource/stack/secrets.go

```diff
@@ -188,3 +192,40 @@ func (c *mapDecrypter) DecryptValue(ciphertext string) (string, error) {
 
 	return plaintext, nil
 }
+
+func (c *mapDecrypter) BulkDecrypt(ciphertexts []string) (map[string]string, error) {
+	// Loop and find the entries that are already cached, then BulkDecrypt the rest
+	secretMap := map[string]string{}
+	var toDecrypt []string
+	if c.cache == nil {
+		// Don't bother searching for the cached subset if the cache is nil
+		toDecrypt = ciphertexts
+	} else {
+		toDecrypt = make([]string, 0)
+		for _, ct := range ciphertexts {
+			if plaintext, ok := c.cache[ct]; ok {
+				secretMap[ct] = plaintext
+			} else {
+				toDecrypt = append(toDecrypt, ct)
+			}
+		}
+	}
+
+	// try and bulk decrypt the rest
+	decrypted, err := c.decrypter.BulkDecrypt(toDecrypt)
+	if err != nil {
+		return nil, err
+	}
+
+	// And add them to the cache
+	if c.cache == nil {
+		c.cache = make(map[string]string)
+	}
+
+	for ct, pt := range decrypted {
+		secretMap[ct] = pt
+		c.cache[ct] = pt
+	}
+
+	return secretMap, nil
+}
```

Comment:
Do we actually need this?

Reply:
I think current data flow is such that no this will never be hit, but it was pretty easy to write and means we won't trip up on a bad method if we refactor things slightly in the future that cause it to start being hit.